### PR TITLE
Fix target not verified issue in event-listeners

### PR DESCRIPTION
**Pull request by pandamicro**
Changes proposed in this pull request:
- Fix target not verified issue in event-listeners https://github.com/cocos-creator/engine/pull/714

> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
> - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/event/event-listeners.js [hunk 1]
```diff
@@ -46,14 +46,18 @@ EventListeners.prototype.invoke = function (event) {
...
+                callingFunc.call(event.currentTarget, event);
+            }
         }
         else {
             endIndex = list.length - 1;
             if (key === cc.Director.EVENT_COMPONENT_UPDATE) {
                 for (i = 1; i <= endIndex; i += 2) {
                     target = list[i];
-                    target.update(event.detail);
+                    if (target.update) {
```
Comment (jareguo) [inline cocos2d/core/event/event-listeners.js]: 这里我不太明白，如果 target 没有 update，怎么会注册到事件列表呢？

Comment (pandamicro) [inline cocos2d/core/event/event-listeners.js]: 这里保护的是 target 被 remove 的情况，在 update 中如果 unschedule 其他 component 的 update，就需要检查了，因为这里的 target 会变成 REMOVE_PLACEHOLDER。

Comment (jareguo) [inline cocos2d/core/event/event-listeners.js]: soga

Comment (jareguo) [inline cocos2d/core/event/event-listeners.js]: 那样的话，不是应该判断 if (target !== REMOVE_PLACEHOLDER) 吗？否则如果 REMOVE_PLACEHOLDER 将来变成 undefine，这里又出错了

Comment (pandamicro) [inline cocos2d/core/event/event-listeners.js]: REMOVE_PLACEHOLDER = undefined 我试了，不行。。。因为数组越界返回的也是 undefined，会导致在检测的时候陷入死循环，所以这里就用 target.update 最保险，不管 PLACEHOLDER 是什么，都不会有 update 函数的

Comment (jareguo) [inline cocos2d/core/event/event-listeners.js]: 那就 REMOVE_PLACEHOLDER = {} 吧，这里没必要增加一次属性查找的开销

Comment (pandamicro) [inline cocos2d/core/event/event-listeners.js]: REMOVE_PLACEHOLDER = {} 也不行，会导致后面 

`hasTarget = target && typeof target === 'object'`

判断错误。

我也不太理解你说的属性查找开销在哪

Comment (jareguo) [inline cocos2d/core/event/event-listeners.js]: 后面应该是先判断 callingFunc !== REMOVE_PLACEHOLDER，这样对移除的对象就可以跳过 callingFunc.call 的检索

**Pull request merged**